### PR TITLE
Add web3-next

### PR DESCRIPTION
**Pull request by Ben-Rey**

### CODE REVIEW

File: src/lib/ConnectMassaWallets/store/accountStore.ts [hunk 1]
```diff
@@ -165,29 +158,12 @@ const accountStore = (
     if (connectedAccount) {
       const currentProvider = get().currentProvider;
       if (!currentProvider) throw new Error('No provider found');
-      const provider = currentProvider;
```
Comment (Thykof) [inline src/lib/ConnectMassaWallets/store/accountStore.ts:168]: ?

File: package.json [hunk 2]
```diff
@@ -40,12 +43,7 @@
     "vite-plugin-svgr": "^3.2.0",
     "zustand": "^4.5.2"
   },
-  "peerDependencies": {
-    "@massalabs/massa-web3": "^4.0.3-dev",
-    "@massalabs/wallet-provider": "^2.0.1-dev",
-    "@types/dot-object": "^2.1.6",
-    "dot-object": "^2.1.5"
-  },
+  "peerDependencies": {},
```
Comment (Thykof) [inline package.json:46]:
```suggestion
```

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 3]
```diff
@@ -22,140 +27,157 @@ function minBigInt(a: bigint, b: bigint) {
...
+      try {
+        maxGas = await gasEstimation(
+          targetFunction,
+          targetAddress,
+          parameter,
+          coins,
+        );
+      } catch (error) {
+        console.error('Gas estimation failed:', error);
+        showToast('error', 'Gas estimation failed.');
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:124]:
```suggestion
        showToast('error', messages.error);
```

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 4]
```diff
@@ -22,140 +27,157 @@ function minBigInt(a: bigint, b: bigint) {
...
+    setIsError(false);
+    setOpId(undefined);
+  }
+
+  async function gasEstimation(
     targetFunction: string,
     targetAddress: string,
-    parameter: number[],
-    messages: ToasterMessage,
+    parameter: Uint8Array,
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:52]:
```suggestion
    parameter: Uint8Array,
    messages: ToasterMessage,
```

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 5]
```diff
@@ -22,140 +27,157 @@ function minBigInt(a: bigint, b: bigint) {
...
+        return;
+      }
+
+      setIsSuccess(true);
+      toast.dismiss(loadingToastId);
+      showToast('success', messages.success, operationId);
+    } catch (error) {
+      console.error('Error during smart contract call:', error);
+      setIsError(true);
+      toast.dismiss(loadingToastId);
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:172]: move it in finally

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 6]
```diff
@@ -22,140 +27,157 @@ function minBigInt(a: bigint, b: bigint) {
...
-          toast.error((t) => (
-            <ToastContent t={t}>
-              <OperationToast
-                isMainnet={isMainnet}
-                title={messages.timeout || Intl.t('steps.failed-timeout')}
-                operationId={operationId}
-              />
-            </ToastContent>
-          ));
+      try {
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:149]: embed try, maybe we can do better

File: src/lib/ConnectMassaWallets/store/accountStore.ts [hunk 7]
```diff
@@ -165,29 +158,12 @@ const accountStore = (
     if (connectedAccount) {
       const currentProvider = get().currentProvider;
       if (!currentProvider) throw new Error('No provider found');
```
Comment (peterjah) [inline src/lib/ConnectMassaWallets/store/accountStore.ts:160]: i think you can also remove this.!

File: src/lib/ConnectMassaWallets/store/accountStore.ts [hunk 8]
```diff
@@ -165,29 +158,12 @@ const accountStore = (
...
-          provider,
-          connectedAccount,
-        ),
-      });
     }
   },
 
   refreshMassaClient: async () => {
     const provider = get().currentProvider;
     if (!provider) return;
```
Comment (peterjah) [inline src/lib/ConnectMassaWallets/store/accountStore.ts:166]: tu remove ?

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx (hunk 6 above)
Comment (Ben-Rey) [reply src/lib/massa-react/hooks/useWriteSmartContract.tsx:150]: Ok I tried a different approach

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 9]
```diff
@@ -22,140 +27,137 @@ function minBigInt(a: bigint, b: bigint) {
...
+  async function getMinimalFee(
+    publicClient: JsonRPCClient,
+    fee?: bigint,
+  ): Promise<bigint> {
+    const minimalFee = await publicClient.getMinimalFee();
+
+    if (!fee) {
+      fee = minimalFee;
+    } else if (fee < minimalFee) {
+      throw new Error('Fee is too low');
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:84]: ?

```suggestion
      return
```

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 10]
```diff
@@ -22,140 +28,151 @@ function minBigInt(a: bigint, b: bigint) {
...
-              <OperationToast
-                isMainnet={isMainnet}
-                title={messages.error}
-                operationId={operationId}
-              />
-            </ToastContent>
-          ));
-          logSmartContractEvents(client, operationId);
-        } else {
-          toast.error((t) => (
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:148]: let's check if we have the same behavior

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 11]
```diff
@@ -22,140 +28,151 @@ function minBigInt(a: bigint, b: bigint) {
...
+      console.error('Error during smart contract call:', error);
+      setIsError(true);
+      showToast(
+        'error',
+        messages.error || Intl.t('steps.failed-timeout'),
+        opId,
+      );
+    } finally {
+      toast.dismiss(loadingToastId);
+      setIsPending(false);
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:173]:
```suggestion
      setIsPending(false);
      setOpIsPending(false);
```

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 12]
```diff
@@ -22,140 +28,151 @@ function minBigInt(a: bigint, b: bigint) {
...
+      loadingToastId = showToast(
+        'loading',
+        messages.pending,
+        operationId,
+        isMainnet,
+      );
+
+      const op = new Operation(publicClient, operationId);
+      const finalStatus = await op.waitSpeculativeExecution();
+
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:150]:
```suggestion

      if (finalStatus === OperationStatus.NotFound) {
        // timeout
      }

```

File: src/lib/massa-react/hooks/useWriteSmartContract.tsx [hunk 13]
```diff
@@ -22,140 +28,151 @@ function minBigInt(a: bigint, b: bigint) {
...
+      }
+
+      setIsSuccess(true);
+      showToast('success', messages.success, operationId);
+    } catch (error) {
+      console.error('Error during smart contract call:', error);
+      setIsError(true);
+      showToast(
+        'error',
+        messages.error || Intl.t('steps.failed-timeout'),
```
Comment (Thykof) [inline src/lib/massa-react/hooks/useWriteSmartContract.tsx:168]:
```suggestion
        messages.error),
```

**Review by Thykof — APPROVED** [review]

**Pull request merged**